tricity/tricity/pipelines.py

```diff
@@ -1,13 +1,43 @@
-# Define your item pipelines here
-#
-# Don't forget to add your pipeline to the ITEM_PIPELINES setting
-# See: https://docs.scrapy.org/en/latest/topics/item-pipeline.html
+import pymongo
+from tricity import settings
+from datetime import datetime
+import hashlib
 
+ 
+class MongoDBPipeline:
+    
+    def set_hash(self,item):
+        result = hashlib.md5(f"{item['price']}{item['url']}".encode('utf-8'))
+        hash_value = result.hexdigest()
+        return hash_value
 
-# useful for handling different item types with a single interface
-from itemadapter import ItemAdapter
+    def set_scrapping_date(self):
+        now = datetime.now().strftime('%Y-%m-%d %H:%M:%S')
+        return now
 
+    def open_spider(self, spider):
+        self.scrapping_date = self.set_scrapping_date()
+        self.client = pymongo.MongoClient(settings.mongodb_uri)
+        db = self.client[settings.mongodb_db]
+        self.collection = db[settings.colection_name]
+        
+        #start with a clean database
+        self.collection.delete_many({})
+
+    def close_spider(self, spider):
+        self.client.close()
 
-class TricityPipeline:
     def process_item(self, item, spider):
-        return item
+        item['price_per_meter'] = str(round(float(item['price']) / float(item['area']), 2))
+        item['hash'] = self.set_hash(item)
+        item['scrapping_date'] = self.scrapping_date
```

Comment:
item['scrapping_date'] - to pole powinno być tylko ustawiane podczas insertu, jeżeli obiekt którego użyjesz do updatu będzie miał to wartość początkowa wartość w bazie zostanie nadpisana, a tego nie chcemy. Przy update jedyną zmianą w bazie danych ma być wartość pola last_seen_date (początkowa wartość tego pola ma być nadana podczas insertu)